### PR TITLE
Improve error message for non-supported session joining

### LINKED ISSUE #5702: tsh join/play fails in proxy record mode on 6.0.0-rc.1
### Description
`tsh join` and `tsh play` fail both in terminal and on the web proxy while in proxy record mode. The only exception is `tsh play` on a local node. 

If this is intended behavior in proxy recording mode, we should fix the "Interact with a cluster using tsh" section in the [Testplan](https://github.com/gravitational/teleport/issues/5654). 

**What happened**:
- `tsh join` gives an error on local and remote nodes: `ERROR: session `[session_id]` not found or it has ended`
- `tsh play` gives an error on remote nodes: `ERROR: open bin/remote/data/log/records/[session_id].tar: no such file or directory
`

**What you expected to happen**:
No errors.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
```bash
 # local node
 > tsh ssh local-node
 # copy session_id from teleport logs or web proxy
 > tsh join [session_id]
 > tsh play [session_id]

 # remote node
 > tsh ssh remote-node
 # copy session_id from teleport logs or web proxy
 > tsh join [session_id]
 > tsh play [session_id]
```

### PR DESCRIPTION
**Pull request by Joerger**
Currently, `tsh join` on a proxy recording session gives an unclear error. Additionally, with the recent change to use `GetSessionTracker` in the `tsh join` implementation, a similar error can occur for non-ssh sessions which still have associated trackers.

I've added a check and better error for both scenarios.

Closes https://github.com/gravitational/teleport/issues/5702

### CODE REVIEW

**Comment by zmb3** [comment]
Is there a better option than `NotImplemented`? I'm used to seeing that for mocks or features that are in progress, but this to me sounds more like an "unsupported" error than not implemented.

**Comment by Joerger** [comment]
> Is there a better option than `NotImplemented`? I'm used to seeing that for mocks or features that are in progress, but this to me sounds more like an "unsupported" error than not implemented.

We do use `NotImplemented` in other places like this, though not very commonly. Examples:
 - https://github.com/gravitational/teleport/blob/19b265048105362c9db34ae7a693f0c193e3b25b/api/client/client.go#L2586
 - https://github.com/gravitational/teleport/blob/19b265048105362c9db34ae7a693f0c193e3b25b/api/client/sessions.go#L65

I think it makes sense here, but we can change it to `BadParameter` or simply `trace.Errorf` if we think that's better.

File: lib/client/api.go [hunk 1]
```diff
@@ -1963,6 +1963,13 @@ func (tc *TeleportClient) Join(ctx context.Context, mode types.SessionParticipan
 		return trace.Wrap(err)
 	}
 
+	// Session joining is not supported in proxy recording mode
+	if recConfig, err := site.GetSessionRecordingConfig(ctx); err != nil {
+		return trace.Wrap(err)
+	} else if services.IsRecordAtProxy(recConfig.GetMode()) {
+		return trace.NotImplemented("session joining is not supported in proxy recording mode")
```
Comment (codingllama) [inline lib/client/api.go:1970]: +1 for trace.BadParameter.

File: lib/client/api.go [hunk 2]
```diff
@@ -1972,6 +1979,10 @@ func (tc *TeleportClient) Join(ctx context.Context, mode types.SessionParticipan
 		return trace.NotFound(notFoundErrorMessage)
 	}
 
+	if session.GetSessionKind() != types.SSHSessionKind {
+		return trace.NotImplemented("session joining is only supported for ssh sessions, not %q sessions", session.GetSessionKind())
```
Comment (codingllama) [inline lib/client/api.go:1983]: Ditto.

**Review by codingllama — APPROVED** [review]

**Review by LKozlowski — APPROVED** [review]

**Pull request merged**